be/src/exprs/vectorized/binary_predicate.cpp

```diff
@@ -76,7 +69,8 @@ class VectorizedNullSafeEqPredicate final : public Predicate {
                 builder.append(false);
             } else {
                 // all not null = value eq
-                builder.append(v1.value(row) == v2.value(row));
+                using CppType = RunTimeCppType<Type>;
```

Comment:
need change it？  `==` seems more readable than `OP::template apply<CppType, CppType, bool>(`

Reply:
The motivation is to specify comparison operator for each type. 

For most PrimitiveType, use operator `==` could satisfies, since `RunTimeCppType` has implement it correctly.

But for some special type, like `object, hll, json`, those `RunTimeCppType` is a pointer, whose operator `==` is meaningless.

As a workaround, these type could specialize `std::equal_to` to solve these problem. So the actual implementation is `std::equal_to<JSONValue>(JSONValue*, JSONValue*)`. The mapping from `CppType` to `CmpType` is implemented by `PredicateCmpType`.

As for readability problem, I could introduce some intermediate struct to improve it.

Reply:
Actually original code of `VectorizedBinaryPredicate` also use the `VectorizedStrictBinaryFunction<OP>::template evaluate<Type, TYPE_BOOLEAN>`,